Update C# Tutorials for RC2

BillWagner opened the pull request:
At the moment, all work except for the microservice. That's waiting for some ASP.NET updates.

Fixes #419

BillWagner committed:
Update Teleprompter sample.

BillWagner committed:
Update Teleprompter sample.

BillWagner committed:
Update webapi client tutorial

BillWagner committed:
Update webapi client tutorial

BillWagner committed:
update console LINQ project for RC2

BillWagner committed:
update console LINQ project for RC2

BillWagner committed:
Update text to explain workaround.

BillWagner committed:
Update text to explain workaround.

BillWagner committed:
Weather Service sample working

Now, to get the Docker instance going.

BillWagner committed:
Weather Service sample working

Now, to get the Docker instance going.

BillWagner committed:
update Dockerfile for RC2 commands

BillWagner committed:
update Dockerfile for RC2 commands

BillWagner commented:
All these now work with the latest CLI and RC2 changes.

/cc @blackdwarf

mairaw commented on docs/tutorials/getting-started-with-csharp/console-teleprompter.md:
````diff
@@ -193,7 +193,14 @@ to add that using statement at the top of file:
 using System.Threading.Tasks;
 ```
 
-Run the sample, and check the output.Now, each single word is printed,
````
Line 17: avoid using here as a link text. Suggestion: You can find the installation instructions on the 
 [Getting started](http://dotnet.github.io/getting-started/) page.

mairaw replied:
Suggestion for line 19:
The descriptions below use -> This tutorial uses

mairaw replied:
Line 44: wouldn't be more accurate to say "creates your application executable"?

mairaw replied:
Line 48: Open that file you your favorite text editor. -> Open that file with your favorite text editor.

mairaw replied:
Line 68: Suggestion: Remove that sentence about the direct link and change the previous paragraph like this
Copy the "[sampleQuotes.txt](https://github.com/dotnet/core-docs/blob/master/docs/tutorials/getting-started-with-csharp/console-teleprompter/sampleQuotes.txt)"
 file from the GitHub repository for this [sample](https://github.com/dotnet/core-docs/tree/master/docs/tutorials/getting-started-with-csharp/console-teleprompter) into your project directory.

mairaw replied:
Line 106: Enumerator Methods -> Enumerator methods

mairaw replied:
Line 126: Var defines -> `var` defines

mairaw replied:
Line 130: which is a `StreamReader` -> which is a `StreamReader` object.

mairaw replied:
nit: Line 312: remove comma after `GetInput()`

mairaw commented on docs/tutorials/getting-started-with-csharp/microservices.md:
```diff
@@ -81,24 +81,23 @@ The template creates six files for you:
-line tools for building and running your applications: dnu and dnx. Dnu is the .NET Execution
-Environment Utility. Dnx is the .NET Execution Engine. 
+of tools from the command line. The `dotnet` command runs the tools necessary
+for .NET development. Each verb execute a different command
```
Each verb execute -> Each verb executes

mairaw commented on docs/tutorials/getting-started-with-csharp/microservices.md:
```diff
@@ -81,24 +81,23 @@ The template creates six files for you:
+`dotnet run`
 
 The default configuration listens to http://localhost:5000. You can open a
 browser and navigate to that page and see a "Hello World!" message.
```
Line 125: It's Main() method -> Its Main() method

mairaw replied:
nit: Line 166: missing space before Run

mairaw commented:
Overall looks good. Left some comments. Didn't follow the steps personally just read them but can do that later on if needed.

BillWagner committed:
Updates from edit comments

BillWagner committed:
Updates from edit comments

BillWagner committed:
Update the Dockerfile to point to the rc2 core image

BillWagner committed:
Update the Dockerfile to point to the rc2 core image

BillWagner commented:
This needs to wait a bit to merge, until there is a Newtonsoft.JSON that works with RC 2. (hopefully soon)

BillWagner committed:
update project.json.

Sadly, NewtonSoft.JSON hasn't been updated for RC 2. This is on hold at
the moment.

BillWagner committed:
update project.json.

Sadly, NewtonSoft.JSON hasn't been updated for RC 2. This is on hold at
the moment.

BillWagner committed:
Update to the prereleaes JSON.NET for RC2

BillWagner committed:
Update to the prereleaes JSON.NET for RC2

BillWagner committed:
update build and docker instructions for RC2 (so close)

BillWagner committed:
update build and docker instructions for RC2 (so close)

BillWagner committed:
update dockerfile for RC2

BillWagner committed:
update dockerfile for RC2

BillWagner committed:
Merge branch 'RC2UpdatesForCSharpTutorials' of https://github.com/BillWagner/core-docs into RC2UpdatesForCSharpTutorials

BillWagner committed:
trying the UseUrls() suggestion.

This is still not working.

BillWagner committed:
Docker working on onbuild image.

BillWagner committed:
Add onbuild dockerfile for reference

BillWagner committed:
Revert "Add onbuild dockerfile for reference"

This reverts commit fa2ad22a38c93c6db6e2782ffc8fd4a20d5a42bb.

BillWagner committed:
Final updates to the article.

BillWagner commented:
Merging, but see issue #462 to add new content (based on priority)
/cc @blackdwarf
